c2rust-transpile/src/rust_ast/mod.rs

```diff
@@ -205,35 +196,3 @@ impl SpanExt for Span {
         get_inner_mut(self)
     }
 }
-
-#[derive(PartialEq, Eq, Debug, Clone)]
-pub struct MySpan {
-    pub lo: u32,
-    pub hi: u32,
-}
-
-impl SpanExt for MySpan {
-    fn is_dummy(&self) -> bool {
-        self.lo == 0 && self.hi == 0
-    }
-
-    fn dummy() -> Self {
-        MySpan { lo: 0, hi: 0 }
-    }
-
-    fn eq(&self, other: &Self) -> bool {
-        self == other
-    }
-
-    fn new(lo: u32, hi: u32) -> Self {
-        MySpan { lo, hi }
-    }
-
-    fn inner(&self) -> (u32, u32) {
-        (self.lo, self.hi)
-    }
-
-    fn inner_mut(&mut self) -> (&mut u32, &mut u32) {
-        (&mut self.lo, &mut self.hi)
-    }
-}
```

Comment:
I'd leave `MySpan` for now in this PR.  Remove it in a separate PR.  I'm not sure what `MySpan` was used for, and although it's related to the `DUMMY_SP` -> `Span::call_site()` change, neither are dependent on each other and it can be removed in a separate PR.

Reply:
Do you think I have nothing better to do with my life than slicing and dicing microPRs?

`MySpan` isn't used either in the code or in the tests, it isn't marked as unused because it's `pub` and has an impl of a pub trait. As far as the compiler is concerned, it is used.

Reply:
> Do you think I have nothing better to do with my life than slicing and dicing microPRs?

No, I don't.  Thank you for all these cleanups, but we don't exactly *need* them at the moment, and it's okay if you want to leave some of these things out if you want to focus on more-involved PRs.  Hopefully someone can eventually get around to these cleanups eventually, and thank you for doing a lot of them already for us, but they aren't the most pressing for us, as we are focused more on getting unsafe-to-safe lifting working, as well as bug fixes.  But importantly, we can't let any of these cleanups break things, so we have to be careful and conservative with them, especially if they aren't in parts of the code that are well-tested.

> `MySpan` isn't used either in the code or in the tests, it isn't marked as unused because it's `pub` and has an impl of a pub trait. As far as the compiler is concerned, it is used.

It could be used sometimes for debugging things, for example.  The point is, I'm not sure how it might be used, so I think it'd be better to wait and ask, as not deleting `MySpan` isn't blocking anything else.  @fw-immunant, who added this code, is now on vacation, so I think it'd be best to hold off on this part of the change.

Reply:
If you'd like, I can split this into another PR for you as this one's quite simple.  I'll force push to your branch to remove that commit and then cherry pick it to a new branch.

Reply:
Split into https://github.com/immunant/c2rust/pull/541.

Reply:
I forced push to remove the `MySpan` removal commit and I'll merge this PR now.